### PR TITLE
Add test for pipedv1's buildPlan method, and remove platform specific config from configv1's GenericApplicationConfig

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does / why we need it**:

- Add test for pipedv1's buildPlan method
- remove platform-specific config from configv1's GenericApplicationConfig
  - and remove platform/stage specific test that is broken by the above changes

**Which issue(s) this PR fixes**:

Part of #4980 

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/config/application.go [hunk 1]
```diff
@@ -60,6 +61,22 @@ type GenericApplicationSpec struct {
...
+	}
+
+	if err := defaults.Set(&aux); err != nil {
+		return err
+	}
+
+	*s = GenericApplicationSpec(aux)
+	return nil
+}
+
```
Comment (Warashi) [inline pkg/config/application.go:79]: Oops, these changes are mistakenly commited. I'll remove these changes.

**Review by t-kikuc — DISMISSED** [review]

File: pkg/app/pipedv1/controller/planner.go [hunk 2]
```diff
@@ -270,7 +273,11 @@ func (p *planner) buildPlan(ctx context.Context, targetDS *deploysource.DeploySo
 		}
 	}
 
-	cfg := targetDS.GenericApplicationConfig
+	cfg := new(config.GenericApplicationSpec)
+	if err := json.NewDecoder(bytes.NewReader(targetDS.ApplicationConfig)).Decode(cfg); err != nil {
+		p.logger.Error("unable to parse application config", zap.Error(err))
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (Warashi) [reply pkg/app/pipedv1/controller/planner.go:280]: OK. I'll move this to configv1, like LoadApplication.

https://github.com/pipe-cd/pipecd/blob/077b7b639a3b128b3163c7023fc64bf6584e8295/pkg/configv1/application.go#L778

[the remark this answers is not in the document]
Comment (Warashi) [reply pkg/app/pipedv1/controller/planner.go]: Fixed on this commit
https://github.com/pipe-cd/pipecd/pull/5238/commits/b9e9261b717b9b10d024865849f3d18345292684

[the remark this answers is not in the document]
Comment (Warashi) [reply pkg/app/pipedv1/controller/planner.go]: I modified again based on face-to-face mtg
https://github.com/pipe-cd/pipecd/pull/5238/commits/85c0a793f88ed29d6086fc2a6257159382314f54

**Review by khanhtc1202 — APPROVED** [review]
LGTM, great step done 💯

**Review by t-kikuc — APPROVED** [review]

**Review by ffjlabo — APPROVED** [review]

**Pull request merged**